cri/v1alpha1/cri_utils_test.go

```diff
@@ -1327,3 +1327,133 @@ func Test_parseResourcesFromCRI(t *testing.T) {
 		})
 	}
 }
+
+func Test_generateMountBindings(t *testing.T) {
+	type args struct {
+		mounts []*runtime.Mount
+	}
+	tests := []struct {
+		name string
+		args args
+		want []string
+	}{
+		{
+			name: "propagation_private test",
+			args: args{
+				mounts: []*runtime.Mount{
+					{
+						ContainerPath:  "container_path",
+						HostPath:       "host_path",
+						Readonly:       true,
+						SelinuxRelabel: true,
+						Propagation:    runtime.MountPropagation_PROPAGATION_PRIVATE,
+					},
+				},
+			},
+			want: []string{"host_path:container_path:ro,Z"},
+		},
+		{
+			name: "propagation_bidirectinal test",
+			args: args{
+				mounts: []*runtime.Mount{
+					{
+						ContainerPath:  "container_path",
+						HostPath:       "host_path",
+						Readonly:       true,
+						SelinuxRelabel: false,
+						Propagation:    runtime.MountPropagation_PROPAGATION_BIDIRECTIONAL,
+					},
+				},
+			},
+			want: []string{"host_path:container_path:ro,rshared"},
+		},
+		{
+			name: "propagation_host_to_container test",
+			args: args{
+				mounts: []*runtime.Mount{
+					{
+						ContainerPath:  "container_path",
+						HostPath:       "host_path",
+						Readonly:       false,
+						SelinuxRelabel: true,
+						Propagation:    runtime.MountPropagation_PROPAGATION_HOST_TO_CONTAINER,
+					},
+				},
+			},
+			want: []string{"host_path:container_path:Z,rslave"},
+		},
+		{
+			name: "no_attrs test",
+			args: args{
+				mounts: []*runtime.Mount{
+					{
+						ContainerPath:  "container_path",
+						HostPath:       "host_path",
+						Readonly:       false,
+						SelinuxRelabel: false,
+					},
+				},
+			},
+			want: []string{"host_path:container_path"},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := generateMountBindings(tt.args.mounts); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("generateMountBindings() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_modifySandboxNamespaceOptions(t *testing.T) {
+	type args struct {
+		nsOpts     *runtime.NamespaceOption
+		hostConfig *apitypes.HostConfig
+	}
+	tests := []struct {
+		name string
+		args args
+		want *apitypes.HostConfig
+	}{
+		{
+			name: "nil test",
+			args: args{
+				nsOpts: nil,
+				hostConfig: &apitypes.HostConfig{
+					PidMode:     namespaceModeHost,
+					IpcMode:     namespaceModeHost,
+					NetworkMode: namespaceModeHost,
+				},
+			},
+			want: &apitypes.HostConfig{
+				PidMode:     namespaceModeHost,
+				IpcMode:     namespaceModeHost,
+				NetworkMode: namespaceModeHost,
+			},
+		},
+		{
+			name: "normal test",
+			args: args{
+				nsOpts: &runtime.NamespaceOption{
+					HostIpc:     true,
+					HostPid:     true,
+					HostNetwork: false,
```

Comment:
More test cases such as `HostNetwork: true` for higher test coverage ?

Reply:
What is the opinion of this comment from Starop? @ZouRui89